src/BuildScriptGenerator/DotNetCore/ProjectFileProviders/ProbeAndFindProjectFileProvider.cs

```diff
@@ -75,14 +77,33 @@ public string GetRelativePathToProjectFile(RepositoryContext context)
                 {
                     azureFunctionsProjects.Add(file);
                 }
+                else if (ProjectFileHelpers.IsAzureBlazorWebAssemblyProject(sourceRepo, file))
+                {
+                    azureBlazorWasmProjects.Add(file);
+                }
             }
 
-            projectFile = GetProject(webAppProjects);
-            if (projectFile == null)
+            // Assumption: some env variable "Oryx_App_type" will be passed to oryx
+            // which will indicate if the app is an azure function type or static type app
+            var oryxAppTypeEnvironmentVar = Environment.GetEnvironmentVariable("Oryx_App_Type");
+            if (projectFile == null && !string.IsNullOrEmpty(oryxAppTypeEnvironmentVar)
+                && oryxAppTypeEnvironmentVar.ToLower().Contains("functions"))
             {
                 projectFile = GetProject(azureFunctionsProjects);
```

Comment:
Having an AppType should not be mandatory for example to build functions projects...only if there are more than 1 projects that were found we use the app type as a hint to filter down the list..so this logic does not look correct to me.

Reply:
I have updated the logic .. have some questions though .. lets discuss tomorrow